src/components/axes/abstractAxis.ts

```diff
@@ -456,30 +456,34 @@ export module Axis {
 
     protected _hideEndTickLabels() {
       var boundingBox = this._element.select(".bounding-box")[0][0].getBoundingClientRect();
-
-      var isInsideBBox = (tickBox: ClientRect) => {
-        return (
-          Math.floor(boundingBox.left) <= Math.ceil(tickBox.left) &&
-          Math.floor(boundingBox.top)  <= Math.ceil(tickBox.top)  &&
-          Math.floor(tickBox.right)  <= Math.ceil(boundingBox.left + this.width()) &&
-          Math.floor(tickBox.bottom) <= Math.ceil(boundingBox.top  + this.height())
-        );
-      };
-
       var tickLabels = this._tickLabelContainer.selectAll("." + AbstractAxis.TICK_LABEL_CLASS);
       if (tickLabels[0].length === 0) {
         return;
       }
       var firstTickLabel = tickLabels[0][0];
-      if (!isInsideBBox(firstTickLabel.getBoundingClientRect())) {
+      if (!_Util.DOM.containedInBoundingBox(boundingBox, firstTickLabel.getBoundingClientRect())) {
         d3.select(firstTickLabel).style("visibility", "hidden");
       }
       var lastTickLabel = tickLabels[0][tickLabels[0].length-1];
-      if (!isInsideBBox(lastTickLabel.getBoundingClientRect())) {
+      if (!_Util.DOM.containedInBoundingBox(boundingBox, lastTickLabel.getBoundingClientRect())) {
         d3.select(lastTickLabel).style("visibility", "hidden");
       }
     }
 
+    // Responsible for hiding any tick labels that break out of the bounding container
+    protected _hideOverflowingTickLabels() {
+      var boundingBox = this._element.select(".bounding-box")[0][0].getBoundingClientRect();
```

Comment:
I think we generally use `selection.node()` instead of `selection[0][0]` to get the element back out.

It might be worth storing off the bounding-box in `AbstractComponent` -- right now this implementation depends on the actual CSS class applied.